src/Json.ts

```diff
@@ -60,3 +60,48 @@ export const parse = (s: string): Either<unknown, Json> => parseJSON(s, identity
  */
 // tslint:disable-next-line: deprecation
 export const stringify = <A>(a: A): Either<unknown, string> => stringifyJSON(a, identity)
+
+// -----------------------------------------------------------------------------
+// destructors
+// -----------------------------------------------------------------------------
+
+/**
+ * Takes a function for each possible Json type and returns the result of the one from the position that matches the given data.
+ *
+ * @category destructors
+ * @example
+ * import * as J from 'fp-ts/Json'
+ *
+ * const typeOf = J.match(
+ *   () => 'null',
+ *   () => 'boolean',
+ *   () => 'number',
+ *   () => 'string',
+ *   () => 'array',
+ *   () => 'object'
+ * )
+ *
+ * assert.deepStrictEqual(typeOf(32.2), 'number')
+ * assert.deepStrictEqual(typeOf(['a', 'b']), 'array')
+ *
+ *  @since 2.10.6
+ */
+export const match: <Z>(
+  isNull: () => Z,
```

Comment:
`is*` is usually used for predicates. I would stick with `on*` to be consistent with match functions of other modules.

Reply:
good point. Changed it.